extensions/eclipse-che-theia-user-preferences/src/browser/che-storage-service.ts

```diff
@@ -7,25 +7,38 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  **********************************************************************/
-import { injectable, inject } from 'inversify';
+import { injectable, inject, postConstruct } from 'inversify';
 import { StorageService } from '@theia/core/lib/browser';
 import { StorageServer } from '../common/storage-server';
 import { ILogger } from '@theia/core/lib/common/logger';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
 
 @injectable()
 export class CheStorageService implements StorageService {
 
     @inject(StorageServer)
     private storageServer: StorageServer;
 
+    @inject(WorkspaceService)
+    protected workspaceService: WorkspaceService;
+
     @inject(ILogger)
     protected logger: ILogger;
 
-    setData<T>(key: string, data?: T): Promise<void> {
+    private initialized: Promise<void>;
+
+    @postConstruct()
+    protected init(): void {
+        this.initialized = this.workspaceService.roots.then(() => {});
+    }
+
+    async setData<T>(key: string, data?: T): Promise<void> {
+        await this.initialized;
```

Comment:
Is it posible to call `await this.workspaceService.roots()` directly?

Reply:
Yes, it is possible, but this variable was introduced in the same way how it done it https://github.com/eclipse-theia/theia/blob/master/packages/workspace/src/browser/workspace-storage-service.ts#L30-L52, which we rebind.

Reply:
Just try, it should work.